RFS/src/testFixtures/java/com/rfs/framework/PreloadedDataContainerOrchestrator.java

```diff
@@ -0,0 +1,285 @@
+                    "].  Please build/pull it first.");
+        }
+        return image.getId();
+    }
+
+    private int getHashCodeOfImagesAndArgs(DockerClient dockerClient, boolean pullIfUnavailable)
+        throws InterruptedException
+    {
+        return new HashHelper(
+                getImageId(dockerClient, baseSourceVersion.imageName, pullIfUnavailable),
+                getImageId(dockerClient, dataLoaderImageName, pullIfUnavailable),
+                generatorContainerArgs).hashCode();
+    }
+
+    private String getImageName() {
+        return PRELOADED_IMAGE_BASE_NAME + baseSourceVersion.prettyName;
+    }
+
+    private static DockerClient createDockerClient() {
+        var config = DefaultDockerClientConfig.createDefaultConfigBuilder().build();
+        var httpClient = new ApacheDockerHttpClient.Builder()
+                .dockerHost(config.getDockerHost())
+                .sslConfig(config.getSSLConfig())
+                .maxConnections(100)
+                .connectionTimeout(Duration.ofSeconds(30))
+                .responseTimeout(Duration.ofSeconds(45))
+                .build();
+
+        return DockerClientImpl.getInstance(config, httpClient);
+    }
+
+    private static Image getExistingImage(DockerClient dockerClient, String imageName, String tag,
+                                          boolean pullIfUnavailable) throws InterruptedException {
+        var fullName = formatFullImageName(imageName, tag);
+        for (var image : dockerClient.listImagesCmd().exec()) {
+            for (String s : image.getRepoTags()) {
+                if (fullName.equals(s)) {
+                    return image;
+                }
+            }
+        }
+        if (pullIfUnavailable) {
+            log.warn("Image not found. Pulling image: " + fullName);
+            dockerClient.pullImageCmd(imageName)
+                    .withTag(tag)
+                    .exec(new PullImageResultCallback())
+                    .awaitCompletion(PULL_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+            return getExistingImage(dockerClient, imageName, tag, false);
+        }
+        return null;
+    }
+
+    @NotNull
+    private static String formatFullImageName(String imageName, String tag) {
+        return imageName + ":" + tag;
+    }
+
+    private void makeNewImage(DockerClient dockerClient, String imageName, String tag)
+            throws IOException, InterruptedException
+    {
+        var imageResponse = dockerClient.inspectImageCmd(baseSourceVersion.imageName).exec();
+        var originalEntrypoint = imageResponse.getConfig().getEntrypoint();
+        var originalCmd = imageResponse.getConfig().getCmd();
+        final var replacementCommand = Streams.concat(
+                        Stream.of("/bin/sh", "-c"),
+                        Stream.of(Streams.concat(
+                                Stream.of("printenv && "),
```

Comment:
I'm not sure I understand what this command is doing, if my guess is right.  We need to be very careful with environment variable output - will this be in the context of the container or the jenkins/github worker?  If the later, it can include security tokens and

Reply:
Thanks - leftover debugging.  Removed